tmail-backend/apps/distributed-es6-backport/src/main/java/com/linagora/tmail/james/app/DistributedServer.java

```diff
@@ -271,6 +280,15 @@ MailboxManager provide(CassandraMailboxManager mailboxManager, KeystoreManager k
         }
     }
 
+    private static class FirebaseListenerDistributedModule extends AbstractModule {
+        @ProvidesIntoSet
+        InitializationOperation registerFirebaseListener(@Named(InjectionKeys.JMAP) RabbitMQEventBus instance, FirebasePushListener firebasePushListener) {
+            return InitilizationOperationBuilder
+                .forClass(FirebasePushListenerRegister.class)
+                .init(() -> instance.register(firebasePushListener));
+        }
+    }
```

Comment:
I do not get what is specific to "Distributed" here.

Any chance there is the exact same code for memory and that this could be monved to the common firebase module?

Reply:
See `PushListener` (also need to explicitly register the listener to the rabbitmq event bus for distributed ones while no need for memory). I tested and see that explicitation is needed (do not get the rationale yet TBH)..